### PR TITLE
Add window/scene information methods

### PR DESCRIPTION
**Pull request by EtaLoop**
Add the following methods : 
- getWindowID
- getWindowList(), getSceneList()
- getNodeList()
- nodeExists()
- setBackgroundColor()

(Not sure about the PR title, please suggest any idea)

### CODE REVIEW

File: src/gepetto_viewer_rerun/client.py [hunk 1]
```diff
@@ -74,27 +74,93 @@ def createScene(self, sceneName: str):
         )
         logger.info(msg)
 
-    def _getSceneIndex(self, sceneName: str):
-        for index, scene in enumerate(self.sceneList):
+    def getWindowList(self):
+        return [scene.name for scene in self.windowList]
```
Comment (nim65s) [inline src/gepetto_viewer_rerun/client.py:78]: this is returning the scene name list, not a window list.

File: src/gepetto_viewer_rerun/client.py [hunk 2]
```diff
@@ -74,27 +74,93 @@ def createScene(self, sceneName: str):
...
+    def getSceneList(self):
+        return [scene.name for scene in self.sceneList]
+
+    def getWindowID(self, name: str):
+        assert isinstance(name, str), "Parameter 'name' must be a string"
+
+        if name not in self.getWindowList():
+            logging.error(f"getWindowID(): Unknown windowName '{name}'.")
+            return ""
+        logger.info("getWindowID() only checks if window exists and return its name.")
```
Comment (nim65s) [inline src/gepetto_viewer_rerun/client.py:89]: maybe this is supposed to provide an index number. 
@florent-lamiraux : are you fine with this method returning a string, or would you expect an int ?

File: src/gepetto_viewer_rerun/client.py [hunk 3]
```diff
@@ -74,27 +74,93 @@ def createScene(self, sceneName: str):
...
+        return name
+
+    def getNodeList(self):
+        entitiesName = []
+        for entity_type in self.entityList:
+            for entity in entity_type:
+                entitiesName.append(entity.name)
```
Comment (nim65s) [inline src/gepetto_viewer_rerun/client.py:96]:
```suggestion
        entitiesName = [entity.name for entity_type in self.entityList for entity in entity_type]
```

File: src/gepetto_viewer_rerun/client.py [hunk 4]
```diff
@@ -74,27 +74,93 @@ def createScene(self, sceneName: str):
...
-                return index
-        return -1
+                return scene
+        return None
+
+    def _getWindow(self, windowName: str):
+        for window in self.windowList:
+            if window.name == windowName:
+                return window
+        return None
```
Comment (nim65s) [inline src/gepetto_viewer_rerun/client.py:114]:
```suggestion
    def _getWindow(self, windowName: str) -> Window | None:
        for window in self.windowList:
            if window.name == windowName:
                return window
```
Returning None is the same as not returning.
Adding return type hints is a bit more explicit than having a `return None` extra line.

And same above. And in general, adding return type hints to other methods would be good.
And we might add some type checker in the tooling, but this is out of scope for this PR

**Review by nim65s — CHANGES_REQUESTED** [review]
I'm not done with the review, but I have to get out of the train :)

File: src/gepetto_viewer_rerun/client.py (hunk 1 above)
Comment (EtaLoop) [reply src/gepetto_viewer_rerun/client.py:78]: Despite the bad variable name (scene instead of window) it still iterate over `self.windowList`

File: src/gepetto_viewer_rerun/client.py (hunk 4 above)
Comment (EtaLoop) [reply src/gepetto_viewer_rerun/client.py:113]: Yes, at some point I planned to specify all return hints for better understanding, so this might be the next PR.

**Comment by florent-lamiraux** [comment]
In `gepetto-viewer-corba`, getWindowId returns a string. For compatibility reasons, I suggests that we keep the same API.

File: src/gepetto_viewer_rerun/client.py (hunk 2 above)
Comment (florent-lamiraux) [reply src/gepetto_viewer_rerun/client.py:89]: In fact method `getWindowId` is deprecated. It returns the string given as input. I suggest that we remove this method.

**Review by florent-lamiraux — CHANGES_REQUESTED** [review]

**Review by florent-lamiraux — APPROVED** [review]
